### ISSUE TITLE
OpenAPI auth credentials reverse order

### ISSUE THREAD

**Issue by Gr3at** [sunscrapers/djoser#679]
This is not a bug, rather a common logic (expected representation convention).
The generated OpenAPI specification schema, presents `username` and `password` fields in reverse (as someone would expect it) order (screenshot attached).

![image](https://user-images.githubusercontent.com/33185243/176148520-affda97b-0d15-413d-9a80-75f323d57a43.png)

I am user drf spectacular to generate the openapi schema.
This might be my faulty configuration, but looking t[o an old related issue](https://github.com/sunscrapers/djoser/issues/41#issuecomment-333806977), the order seems to also be `password` before the `username`.

**Comment by RahulGupta237** [comment]
HI i didnot understand please where am i go to changes for this issue

**Comment by insspb** [comment]
This is not issue. If you use djoser's universal serializer: The reason of such behaviour is that `password` field is always exist, `username` is optional.

If you use own serializer: Rewrite order on your side.

If you use djosers serializer, but want to change rendering in spectacular, you can create custom serializer, without logic, just with fields definition, and use spectacular extend_schema for overwrite.

**Comment by Gr3at** [comment]
> This is not issue. If you use djoser's universal serializer: The reason of such behaviour is that `password` field is always exist, `username` is optional.
> 
> If you use own serializer: Rewrite order on your side.
> 
> If you use djosers serializer, but want to change rendering in spectacular, you can create custom serializer, without logic, just with fields definition, and use spectacular extend_schema for overwrite.

It's been quite some time, since I had this UX (imo) issue.

So, if I understand it correctly, what you suggest is the following: 
```python
# serializers.py
class CustomAuthSerializer(serializers.Serializer):
    username = serializers.CharField(min_length=3, max_length=20)
    password = serializers.CharField(min_length=8, max_length=50, write_only=True)
```
```python
# views.py
# override the default login api view, post method to add the following
@extend_schema(description="Authentication token retrieval endpoint.", request=[CustomAuthSerializer])
```

**Comment by insspb** [comment]
@Gr3at Yes, something like this. i will not help on concrete @extend... decorator and parameters, as spectacular has several options, not only extend_schema. But conceptually yes, if you want to reorder fields in controlled manner, the best way to create custom serializer for documentation only.
